### PR TITLE
qemu: libiscsi support

### PR DESCRIPTION
**Pull request by misuzu**
###### Motivation for this change
Support for booting directly from iSCSI.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Think this'll need a rebase.

**Comment by misuzu** [comment]
> Think this'll need a rebase.

Done!

**Comment by risicle** [comment]
Builds on macos 10.14.

@ofborg build qemu

**Review by risicle — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Please target master this is not a mass rebuild.

File: pkgs/development/libraries/libiscsi/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, fetchFromGitHub, autoreconfHook }:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libiscsi/default.nix:2]:
```suggestion
{ stdenv, fetchFromGitHub, autoreconfHook }:

stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/libiscsi/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  pname = "libiscsi";
+  version = "1.19.0";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libiscsi/default.nix:5]:
```suggestion
  version = "1.19.0";

  src = fetchFromGitHub {
```

File: pkgs/development/libraries/libiscsi/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    owner = "sahlberg";
+    repo = "libiscsi";
+    rev = version;
+    sha256 = "0ajrkkg5awmi8m4b3mha7h07ylg18k252qprvk1sgq0qbyd66zy7";
+  };
+  nativeBuildInputs = [ autoreconfHook ];
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libiscsi/default.nix:12]:
```suggestion
  };

  nativeBuildInputs = [ autoreconfHook ];

  meta = with stdenv.lib; {
```

**Comment by misuzu** [comment]
> Please target master this is not a mass rebuild.

Done!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 95274` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
11 packages marked as broken and skipped:
</summary>
<ul>
<li>alpine-make-vm-image</li>
<li>cloud-init</li>
<li>cloud-utils</li>
<li>multibootusb</li>
<li>python37Packages.guestfs</li>
<li>python38Packages.guestfs</li>
<li>python39Packages.guestfs</li>
<li>qemu_xen</li>
<li>qemu_xen-light</li>
<li>qemu_xen_4_10</li>
<li>qemu_xen_4_10-light</li>
</ul>
</details>
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>cot: log <a href="https://termbin.com/dort">https://termbin.com/dort</a></li>
<li>(python38Packages.cot): log was empty</li>
<li>python37Packages.cot: log <a href="https://termbin.com/59tw">https://termbin.com/59tw</a></li>
<li>python39Packages.cot: log <a href="https://termbin.com/2jf1">https://termbin.com/2jf1</a></li>
  </ul>
</details>
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>libiscsi</li>
<li>out-of-tree</li>
<li>qemu</li>
<li>qemu-utils</li>
<li>qemu_kvm</li>
<li>qemu_test</li>
</ul>

```diff
</details>
diff --git a/pkgs/development/python-modules/cot/default.nix b/pkgs/development/python-modules/cot/default.nix
index 89e26545d79..0c2bb961290 100644
--- a/pkgs/development/python-modules/cot/default.nix
+++ b/pkgs/development/python-modules/cot/default.nix
@@ -35,9 +35,10 @@ buildPythonPackage rec {
     "TestQCOW2"
     "TestRAW"
     "TestVMDKConversion"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "test_serial_fixup_invalid_host"
   ];
 
-
   # argparse is part of the standardlib
   prePatch = ''
     substituteInPlace setup.py --replace "'argparse'," ""
```

**Comment by misuzu** [comment]
> This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
> If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.
> 
> Result of `nixpkgs-review pr 95274` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 
>  11 packages marked as broken and skipped:
> 4 packages failed to build and are new build failures:
> * cot: log https://termbin.com/dort
> * (python38Packages.cot): log was empty
> * python37Packages.cot: log https://termbin.com/59tw
> * python39Packages.cot: log https://termbin.com/2jf1
> 
>  6 packages built:

It builds just fine on my machine.
```
$ uname -sr
Darwin 19.6.0
$ nixpkgs-review pr 95274
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/95274/head:refs/nixpkgs-review/1
$ git worktree add /Users/misuzu/.cache/nixpkgs-review/pr-95274-3/nixpkgs 5d9a3e2e5a05b979d3953eeeb1a8282b241a1f53
Preparing worktree (detached HEAD 5d9a3e2e5a0)
Updating files: 100% (24008/24008), done.
HEAD is now at 5d9a3e2e5a0 cargo-c: 0.7.0 -> 0.7.1
$ git merge --no-commit c5dd597870e90f2bf9a03c5a61517eab5c145740
Auto-merging pkgs/top-level/all-packages.nix
Auto-merging pkgs/applications/virtualization/qemu/default.nix
Automatic merge went well; stopped before committing as requested
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /Users/misuzu/.cache/nixpkgs-review/pr-95274-3/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/95274

11 packages marked as broken and skipped:
alpine-make-vm-image cloud-init cloud-utils multibootusb python37Packages.guestfs python38Packages.guestfs python39Packages.guestfs qemu_xen qemu_xen-light qemu_xen_4_10 qemu_xen_4_10-light

9 packages built:
cot libiscsi out-of-tree python37Packages.cot python39Packages.cot qemu qemu-utils qemu_kvm qemu_test
```

**Comment by SuperSandro2000** [comment]
> It builds just fine on my machine.

Do you have sandbox enabled?

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by misuzu** [comment]
> > It builds just fine on my machine.
> 
> Do you have sandbox enabled?

Yes. I have multi-user nix with explicitly enabled sandbox.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 95274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages_hardkernel_4_14.virtualbox</li>
<li>linuxPackages_hardkernel_latest.virtualbox</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixosTest-test</li>
</ul>
</details>
<details>

  <summary>14 packages failed to build and are new build failures:</summary>
  <ul>
<li>aqemu: log <a href="https://termbin.com/8ga2">https://termbin.com/8ga2</a></li>
<li>linuxPackages-libre.virtualbox: log was empty</li>
<li>linuxPackages_5_10.virtualbox: log <a href="https://termbin.com/mkq3">https://termbin.com/mkq3</a></li>
<li>(linuxPackages_latest.virtualbox): log was empty</li>
<li>linuxPackages_latest-libre.virtualbox: log <a href="https://termbin.com/kbi2">https://termbin.com/kbi2</a></li>
<li>linuxPackages_latest_hardened.virtualbox: log <a href="https://termbin.com/254a">https://termbin.com/254a</a></li>
<li>linuxPackages_latest_xen_dom0.virtualbox: log <a href="https://termbin.com/t7vz">https://termbin.com/t7vz</a></li>
<li>linuxPackages_latest_xen_dom0_hardened.virtualbox: log <a href="https://termbin.com/oe5f">https://termbin.com/oe5f</a></li>
<li>linuxPackages_lqx.virtualbox: log <a href="https://termbin.com/p8jb">https://termbin.com/p8jb</a></li>
<li>linuxPackages_zen.virtualbox: log <a href="https://termbin.com/11s2">https://termbin.com/11s2</a></li>
<li>qemu_xen: log was empty</li>
<li>(qemu_xen_4_10): log was empty</li>
<li>qemu_xen-light: log was empty</li>
<li>(qemu_xen_4_10-light): log was empty</li>
  </ul>
</details>
<details>
<summary>
34 packages built:
</summary>
<ul>
<li>alpine-make-vm-image</li>
<li>cloud-init</li>
<li>cloud-utils</li>
<li>cot (python38Packages.cot)</li>
<li>libguestfs</li>
<li>libiscsi</li>
<li>linuxPackages.virtualbox (linuxPackages_5_4.virtualbox)</li>
<li>linuxPackages_4_14.virtualbox</li>
<li>linuxPackages_4_19.virtualbox</li>
<li>linuxPackages_4_4.virtualbox</li>
<li>linuxPackages_4_9.virtualbox</li>
<li>linuxPackages_5_9.virtualbox</li>
<li>linuxPackages_hardened.virtualbox</li>
<li>linuxPackages_testing_bcachefs.virtualbox</li>
<li>linuxPackages_xen_dom0.virtualbox</li>
<li>linuxPackages_xen_dom0_hardened.virtualbox</li>
<li>multibootusb</li>
<li>open-watcom-bin</li>
<li>out-of-tree</li>
<li>python37Packages.cot</li>
<li>python37Packages.guestfs</li>
<li>python38Packages.guestfs</li>
<li>python39Packages.cot</li>
<li>python39Packages.guestfs</li>
<li>qemu</li>
<li>qemu-utils</li>
<li>qemu_kvm</li>
<li>qemu_test</li>
<li>qtemu</li>
<li>vagrant</li>
<li>virtualbox</li>
<li>virtualboxHardened</li>
<li>virtualboxHeadless</li>
<li>virtualboxWithExtpack</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

virtualboxHeadless:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unused --replace got detected:
```
substituteStream(): WARNING: pattern 'PSHCLFORMATDATA' doesn't match anything in file 'src/VBox/HostServices/SharedClipboard/VBoxSharedClipboardSvc-x11-stubs.cpp'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by misuzu** [comment]
> This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
> If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.
> 
> Result of `nixpkgs-review pr 95274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
>  2 packages marked as broken and skipped:
>  1 package blacklisted:
> 14 packages failed to build and are new build failures:
>  34 packages built:
> The following issues got detected with the above build packages.
> Please fix at least the ones listed with your changed packages:
> 
> virtualboxHeadless:

[aqemu](https://hydra.nixos.org/eval/1642710?filter=aqemu&compare=1642471&full=#tabs-still-fail) and [virtualbox](https://hydra.nixos.org/eval/1642710?filter=virtualbox&compare=1642471&full=#tabs-still-fail) packages are already broken on master and unrelated to this change.

**Pull request merged**